### ISSUE TITLE
setting kubernetes service version to latest release (1.2.2) gives error

### ISSUE THREAD

**Issue by eddytruyen** [mbruzek/layer-k8s#27]
Commands:

```
juju bootstrap
juju deploy trusty/etcd
juju deploy cs:~containers/trusty/kubernetes-0
juju get kubernetes
juju set kubernetes version=1.2.2
 juju add-relation kubernetes etcd
```

debug-log output:

```
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40   File "/var/lib/juju/agents/unit-kubernetes-0/charm/reactive/k8s.py", line 153, in master
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40     compose.up('kubelet')
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/docker/compose.py", line 35, in up
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40     self.run(cmd)
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/docker/compose.py", line 75, in run
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40     out = check_output(split(cmd))
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40   File "/usr/lib/python3.4/subprocess.py", line 620, in check_output
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40     raise CalledProcessError(retcode, process.args, output=output)
unit-kubernetes-0[7748]: 2016-04-13 22:22:50 INFO unit.kubernetes/0.etcd-relation-joined logger.go:40 subprocess.CalledProcessError: Command '['docker-compose', 'up', '-d', 'kubelet']' returned non-zero exit status 1
```

I think there is a problem with file `/var/lib/juju/agents/unit-kubernetes-0/charm/files/kubernetes\docker-compose.yml`
[docker-compose.zip](https://github.com/mbruzek/layer-k8s/files/218087/docker-compose.zip)

**Comment by mbruzek** [comment]
Hello @eddytruyen thanks for filing the issue. We have not tested with 1.2.2 yet and I suspect the new release may have changed some things we are not yet up-to-date with. 

Regarding the compose error, I believe the way compose is delivered has changed. It used to be a debian package, now it is delivered via pip. The docker layer contains the version of compose used using the wheelhouse.txt file. If you notice that a specific version does or does not work we can pin versions that work and fix the versions that do not work.

**Comment by mbruzek** [comment]
@eddytruyen according to the [docker kubernetes docs](http://kubernetes.io/docs/getting-started-guides/docker/) the name of the container has changed from `hyperkube` to `hyperkube-amd64`. This will require our code to change to use the new container name. I will be working on this solution.

**Comment by lazypower** [comment]
closed with #40
